From 662514a12a5599aa96c95825e470623215d657f9 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 2 Sep 2024 12:33:41 +0200
Subject: [PATCH 1220/1695] dt-bindings: scmi: add scmi clock flags

Theses values could be used to fill flags property of an SCMI clock in
device-tree:

clock@0 {
    reg = <0>;
    clocks = <&clk 0>;
    domain-name = "ck_flexgen_11";
    flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
              SCMI_CLOCK_ALLOW_SET_RATE)>;
};

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c7d309b6b58f4831d5c7fdabcf2c12bfe8a79f9e dt-bindings: scmi: add scmi clock flags]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I48803b62d7485bb5fee0664d078fe6eefdad48ce
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/401855
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/include/dt-bindings/scmi/scmi-clock.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 core/include/dt-bindings/scmi/scmi-clock.h

diff --git a/core/include/dt-bindings/scmi/scmi-clock.h b/core/include/dt-bindings/scmi/scmi-clock.h
new file mode 100644
index 000000000..2c8fd550a
--- /dev/null
+++ b/core/include/dt-bindings/scmi/scmi-clock.h
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause */
+/*
+ * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_SCMI_SCMI_CLOCK_H_
+#define _DT_BINDINGS_SCMI_SCMI_CLOCK_H_
+
+#define SCMI_CLOCK_DEFAULT_ENABLED	0x1
+#define SCMI_CLOCK_ALLOW_SET_RATE	0x2
+#define SCMI_CLOCK_MASK			GENMASK_32(1, 0)
+
+#endif /* _DT_BINDINGS_SCMI_SCMI_CLOCK_H_ */
-- 
2.47.2

